batch/src/main/java/org/openmrs/analytics/FhirEtl.java

```diff
@@ -0,0 +1,161 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.openmrs.analytics;
+
+import ca.uhn.fhir.context.FhirContext;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.coders.CannotProvideCoderException;
+import org.apache.beam.sdk.options.Default;
+import org.apache.beam.sdk.options.Description;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.options.Validation.Required;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.DomainResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A Beam pipeline for reading FHIR resources from OpenMRS and pushing them into a data warehouse.
+ */
+public class FhirEtl {
+  private static final Logger log = LoggerFactory.getLogger(FhirEtl.class);
+
+  /**
+   * Options supported by {@link FhirEtl}.
+   */
+  public interface FhirEtlOptions extends PipelineOptions {
+
+    /**
+     * By default, this reads from the MySQL DB `openmrs` at the default port on localhost.
+     */
+    @Description("OpenMRS server URL")
+    @Required
+    @Default.String("http://localhost:8080/openmrs")
+    String getServerUrl();
+    void setServerUrl(String value);
+
+    @Description("Comma separated list of resource and search parameters to fetch; in its simplest "
+        + "form this is a list of resources, e.g., `Patient,Encounter,Observation` but more "
+        + "complex search paths are possible too, e.g., `Patient?name=Susan`.")
+    @Default.String("Patient,Encounter,Observation")
+    String getSearchList();
+    void setSearchList(String value);
+
+    @Description("The number of resources to be fetched in one API call.")
+    @Default.Integer(10)
+    int getBatchSize();
+    void setBatchSize(int value);
+
+    @Description("JSESSIONID cookie value to circumvent OpenMRS server authentication.")
```

Comment:
Would we like to consider using BasicAuth? I use it with the HAPI client like this: https://github.com/pmanko/openmrs-fhir-analytics/blob/d54a264f5756081a4ad86a89e2faa40c02970537/src/main/java/org/openmrs/FhirEventWorker.java#L56